src/pages/Main.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState, useEffect } from 'react';
+import Carousel from 'react-multi-carousel';
+import Header from 'components/Header';
+import CardItem from 'components/CardItem';
+import styled from 'styled-components';
+import 'react-multi-carousel/lib/styles.css';
+
+import { usePostContext } from 'contexts/Post';
+
+const responsive = {
+  desktop: {
+    breakpoint: { max: 3000, min: 1024 },
+    items: 1,
+    paritialVisibilityGutter: 60,
+  },
+  tablet: {
+    breakpoint: { max: 1024, min: 464 },
+    items: 1,
+    paritialVisibilityGutter: 50,
+  },
+  mobile: {
+    breakpoint: { max: 464, min: 0 },
+    items: 1,
+    paritialVisibilityGutter: 30,
+  },
+};
+
+const Main = () => {
+  const [nextSlide, setNextSlide] = useState(0);
+  const { state, actions } = usePostContext();
+  const reversePosts = [...state.posts].reverse();
```

Comment:
```suggestion
  const reversePosts = useMemo(() => state.posts.reverse(), [state.posts])
```
reverse함수는 리렌더링시마다 배열을 도는 비용이 들어 메모이제이션 해주면 조금이나마 성능상 이점이 있을꺼같아요!

Reply:
reverse가 원본 배열을 뒤집는 이슈가 있어서 
```
useMemo(() => {
  const newPosts = [...state.posts];
  newPosts.reverse();
  return newPosts;
}, [state.posts]);
```
가 되어야 할 거 같아요

Reply:
오오! 안그래도 이부분 코드가 마음에 걸렸는데.. 감사합니다!